evap/staff/views.py

```diff
@@ -91,6 +90,100 @@ def semester_view(request, semester_id):
 
 
 @staff_required
+def semester_course_operation(request, semester_id):
+    semester = get_object_or_404(Semester, id=semester_id)
+    raise_permission_denied_if_archived(semester)
+
+    course_ids = request.GET.getlist('course')
+    operation = request.GET.get('operation')
+    if operation not in ['revertToNew', 'prepare', 'reenableLecturerReview', 'approve', 'publish', 'unpublish']:
+        messages.error(request, _("Unsupported operation: ") + str(operation))
+        return custom_redirect('evap.staff.views.semester_view', semester_id)
+    courses = [get_object_or_404(Course, id=course_id) for course_id in course_ids]
+
+    if request.method == 'POST':
+        if operation == 'revertToNew':
+            for course in courses:
+                course.revert_to_new()
+                course.save()
+            messages.success(request, __("Successfully reverted %(courses)d course to new.",
+                "Successfully reverted %(courses)d courses to new.", len(courses)) % {'courses': len(courses)})
+
+        elif operation == 'prepare' or operation == 'reenableLecturerReview':
+            for course in courses:
+                course.ready_for_contributors()
+                course.save()
+            messages.success(request, __("Successfully enabled %(courses)d course for lecturer review.",
+                "Successfully enabled %(courses)d courses for lecturer review.", len(courses)) % {'courses': len(courses)})
+            try:
+                EmailTemplate.get_review_template().send_to_users_in_courses(courses, ['editors'])
+            except Exception:
+                messages.error(request, _("An error occured when sending the notification emails to the lecturers."))
+
+        elif operation == 'approve':
+            approved = 0
+            for course in courses:
+                if course.has_enough_questionnaires:
```

Comment:
we can't just deactivate the checkboxes because in some states approving is just one of the possible operations. should i already remove these courses from the list on the confirmation page and display the error message there?